feat: add C implementation for `stats/base/dists/normal/entropy`

manvith2003 opened the pull request:
Resolves #3772

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/normal/entropy along with relevant docs, tests, examples and   benchmarks.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves  [[RFC]: Add C implementation for @stdlib/stats/base/dists/normal/entropy   #3772
](https://github.com/stdlib-js/stdlib/issues/3772)

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

manvith2003 committed:
feat: add C implementation for stats/base/dists/normal/entropy

manvith2003 committed:
feat: add C implementation for stats/base/dists/normal/entropy

manvith2003 committed:
docs: fix comments in the documentation

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into normal_entropy

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/normal/entropy/src/main.c:
```diff
@@ -0,0 +1,61 @@
+
+#include "stdlib/stats/base/dists/normal/entropy.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/log.h"
```
@manvith2003 We should use `#include "stdlib/math/base/special/ln.h"` when the base is `e`.

See: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/ln

manvith2003 replied:
@Planeshifter  fixed the issue

Planeshifter replied:
@manvith2003 Awesome, thank you!

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/normal/entropy/src/main.c:
```diff
@@ -0,0 +1,61 @@
+*
+* @example
+* double y = stdlib_base_normal_entropy( 0.0, 0.0 );
+* // returns NaN
```
Let's just keep one example here:

```suggestion
* @example
* double y = stdlib_base_normal_entropy( 0.0, 1.0 );
* // returns ~1.419
```

Planeshifter reviewed (changes requested):
Looks good overall. However, we should use `math/base/special/ln` for the natural logarithm. That should be updated before merging.

manvith2003 committed:
docs(main.c): update documentation to resolve issue

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/normal/entropy/manifest.json:
```diff
@@ -0,0 +1,88 @@
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/special/ln",
+        "@stdlib/math/base/special/sqrt",
```
@manvith2003 I believe `sqrt` is not used and can be removed in these three configs. Please double-check.

manvith2003 replied:
@Planeshifter  resolved it

manvith2003 committed:
docs: resolved the issue in mainfest.json

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter reviewed (approved):
Thanks for the update. Let's get your PR merged...
